src/rewriter.coffee

```diff
@@ -685,6 +685,8 @@ exports.Rewriter = class Rewriter
           tokens.splice i, 1, @indentation()...
           return 1
         if @tag(i + 1) in EXPRESSION_CLOSE
+          if token[1] is ';' and @tag(i + 1) is 'OUTDENT'
```

Comment:
By storing a reference to the `;` token preceding an `OUTDENT`, we can preserve both its comments and location data